plugins/hydra_nevergrad_sweeper/tests/test_nevergrad_sweeper_plugin.py

```diff
@@ -5,43 +5,71 @@
 from typing import Any
 
 import nevergrad as ng
-import pytest
+import pytest  # type: ignore
+from hydra.core.override_parser.overrides_parser import OverridesParser
 from hydra.core.plugins import Plugins
 from hydra.plugins.sweeper import Sweeper
 from hydra.test_utils.test_utils import TSweepRunner, chdir_plugin_root
 from omegaconf import DictConfig, OmegaConf
 
 from hydra_plugins.hydra_nevergrad_sweeper import core
+from hydra_plugins.hydra_nevergrad_sweeper.nevergrad_sweeper import (
+    NevergradSweeper,  # type: ignore
+)
 
 chdir_plugin_root()
 
 
 def test_discovery() -> None:
-    assert core.NevergradSweeper.__name__ in [
+    assert NevergradSweeper.__name__ in [
         x.__name__ for x in Plugins.instance().discover(Sweeper)
     ]
 
 
 @pytest.mark.parametrize(  # type: ignore
-    "description,param_cls,value_cls",
+    "params,param_cls,value_cls",
+    [
+        ([1, 2, 3], ng.p.Choice, int),
+        (["1", "2", "3"], ng.p.Choice, str),
+        ({"lower": 1, "upper": 2, "log": True}, ng.p.Log, float),
+        ({"lower": 1, "upper": 2}, ng.p.Scalar, float),
+        ({"lower": 1, "upper": 12, "integer": True}, ng.p.Scalar, int),
+        ({"lower": 1, "upper": 12, "log": True, "integer": True}, ng.p.Log, int),
+    ],
+)
+def test_get_nevergrad_parameter_from_config(
+    params: str, param_cls: Any, value_cls: Any
+) -> None:
+    param = core.get_nevergrad_parameter(params)
+    assert isinstance(param, param_cls)
+    if param_cls is not str:
+        assert isinstance(param.value, value_cls)
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "override,param_cls,value_cls",
     [
-        ("blu,blublu", ng.p.Choice, str),
-        (["blu", "blublu"], ng.p.Choice, str),
-        ("0,1,2", ng.p.TransitionChoice, float),
-        ([0, 1, 2], ng.p.TransitionChoice, float),
-        ("int:0,1,2", ng.p.TransitionChoice, int),
-        ("str:0,1,2", ng.p.Choice, str),
-        ("0.0,12.0,2.0", ng.p.Choice, float),
-        ("int:1:12", ng.p.Scalar, int),
-        ("1:12", ng.p.Scalar, float),
-        ("log:0.01:1.0", ng.p.Log, float),
-        ("blublu", str, str),
+        ("key=choice(1,2)", ng.p.Choice, int),
```

Comment:
can you change this to:
`"override,expected"`
e.g:
`key=choice(1,2)", ng.p.Choice(items=[1,2])`
Or however you represetnt a choice between 1 and 2 in Nevergrad?
going through the class here only makes the test more obscure, I think it's not adding any value.

Reply:
sounds good!